### PR TITLE
Add support for remote DOCKER_HOST

### PR DESCRIPTION
**Pull request by str3tch**
Hi,

On our corporate network, we have a remote docker machine (defined in the DOCKER_HOST env var), rather than being able to spin up containers locally. This means that when you use Dockerizepip, it fails when trying to mount the volume (can't mount local filesystem to a remote docker server). So.. the fix is that instead of mounting the local folder, create a docker volume, copy the files into it, mount that to the container, do the pip install magic, then copy the files back off the container.

I've got this working, see this PR. I have run the tests and they pass, but I can't get the BATS tests to run on my Mac. so not quite there. 

I am willing to beef up the testing if it's required.. but first I'd like feedback that this PR is something that you're actually interested in. 

Let me know..

cheers!
Str3tch

### CODE REVIEW

**Comment by dschep** [comment]
Could you run `yarn format` to get the CI build passing? For the testing, feel free to implement your test in the `tape` based tests in `test.js` instead of worrying about getting bats running.

File: README.md [hunk 1]
```diff
@@ -72,6 +72,23 @@ It is important that the host of your private repositories has already been adde
...
+
+Additionally, if you are in a corporate network and there is a proxy, you can define this in the `proxy` option. 
+This will set the http and https proxy inside the container.
+
+```yaml 
+custom:
+  pythonRequirements:
+    dockerizePip: true
+    dockerRemoteHost: true
+    proxy: http://yourproxy:8080
```
Comment (dschep) [inline README.md]: I just merged #271. Please remove the proxy option as users can use that feature to pass proxies in.

File: README.md [hunk 2]
```diff
@@ -72,6 +72,23 @@ It is important that the host of your private repositories has already been adde
...
+it uses a Docker volume and copies the files across.
+
+Additionally, if you are in a corporate network and there is a proxy, you can define this in the `proxy` option. 
+This will set the http and https proxy inside the container.
+
+```yaml 
+custom:
+  pythonRequirements:
+    dockerizePip: true
+    dockerRemoteHost: true
```
Comment (dschep) [inline README.md]: Should the plugin maybe not actually have an option for this, and automatically do all the volume stuff id `DOCKER_HOST` is set?

File: README.md (hunk 1 above)
Comment (str3tch) [reply README.md]: Ok great, will do

**Comment by str3tch** [comment]
Sorry for delay on getting this sorted, smashed at work at the mo. 

I did do a yarn format but still seems to complain.. will try and take a look next week and get some tests together.

**Comment by jamesryanbell** [comment]
Is there any update on this? 😃

**Comment by alexfre** [comment]
Any news on this? It's required to run on CI engines like CircleCI which utilize remote docker machine

**Comment by str3tch** [comment]
Hey guys, apologies.. been snowed under at work and have a newborn at home - pretty hectic! I’ll try and get to it, but it’s not going to be asap .. if anyone wants to tidy it up, not much to do to get it passing another review.

**Comment by Dunedan** [comment]
Any update on this? I'd much appreciate being able to run the packaging on a remote docker host as well.

**Comment by pgrzesik** [comment]
Hey @str3tch - it's been a long time since this PR was proposed. I'm going to close it, if you feel like the issue is valid, please open a new issue or a new PR against the latest main branch. Thanks 🙇

**Pull request closed without merging**